webapp/src/components/dot_menu/dot_menu.tsx

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ComponentProps, useState} from 'react';
+import {PrimaryButton} from 'src/components/buttons';
+import styled, {css} from 'styled-components';
+
+import Dropdown from './dropdown';
+
+export const DotMenuButton = styled.div<{ isActive: boolean }>`
+    display: inline-flex;
+    padding: 0;
+    border: none;
+    border-radius: 4px;
+    width: 28px;
+    height: 28px;
+    align-items: center;
+    justify-content: center;
+    fill: rgba(var(--center-channel-color-rgb), 0.56);
+    cursor: pointer;
+
+    color: ${(props) => (props.isActive ? 'var(--button-bg)' : 'rgba(var(--center-channel-color-rgb), 0.56)')};
+    background-color: ${(props) => (props.isActive ? 'rgba(var(--button-bg-rgb), 0.08)' : 'transparent')};
+
+    &:hover {
+        color: ${(props) => (props.isActive ? 'var(--button-bg)' : 'rgba(var(--center-channel-color-rgb), 0.56)')};
+        background-color: ${(props) => (props.isActive ? 'rgba(var(--button-bg-rgb), 0.08)' : 'rgba(var(--center-channel-color-rgb), 0.08)')};
+    }
+`;
+
+export const DropdownMenu = styled.div`
+    display: flex;
+    flex-direction: column;
+
+    width: max-content;
+    min-width: 16rem;
+    text-align: left;
+    list-style: none;
+
+    padding: 10px 0;
+    font-family: Open Sans;
+    font-style: normal;
+    font-weight: normal;
+    font-size: 14px;
+    line-height: 20px;
+    color: var(--center-channel-color);
+
+    background: var(--center-channel-bg);
+    border: 1px solid rgba(var(--center-channel-color-rgb), 0.16);
+    box-shadow: 0px 8px 24px rgba(0, 0, 0, 0.12);
+    border-radius: 4px;
+
+    z-index: 12;
+`;
+
+type DotMenuProps = {
+    children: React.ReactNode;
+    icon: JSX.Element;
+    dotMenuButton?: typeof DotMenuButton | typeof PrimaryButton;
+    dropdownMenu?: typeof DropdownMenu;
+    title?: string;
+    disabled?: boolean;
+    className?: string;
+    isActive?: boolean;
+    closeOnClick?: boolean;
+};
+
+type DropdownProps = Omit<ComponentProps<typeof Dropdown>, 'target' | 'children' | 'isOpen'>;
+
+const DotMenu = ({
+    children,
+    icon,
+    title,
+    className,
+    disabled,
+    isActive,
+    closeOnClick = true,
+    dotMenuButton: MenuButton = DotMenuButton,
+    dropdownMenu: Menu = DropdownMenu,
+    ...props
+}: DotMenuProps & DropdownProps) => {
+    const [isOpen, setOpen] = useState(false);
+    const toggleOpen = () => {
+        setOpen(!isOpen);
+    };
+
+    const button = (
+
+        // @ts-ignore
+        <MenuButton
+            title={title}
+            isActive={(isActive ?? false) || isOpen}
+            onClick={(e: MouseEvent) => {
+                e.preventDefault();
+                e.stopPropagation();
+                toggleOpen();
+            }}
+            onKeyDown={(e: KeyboardEvent) => {
```

Comment:
Any reason for not using key up? Just feels more natural to me and it avoids spamming.